parser/proto.y

```diff
@@ -488,16 +488,26 @@ listLiteral : '[' listElements ']' {
 		$$ = ast.NewArrayLiteralNode($1, nil, nil, $2)
 	}
 	| '[' error ']' {
-		$$ = nil
+		$$ = ast.NewArrayLiteralNode($1, nil, nil, $3)
 	}
 
 listElements : listElement {
-		$$ = &valueSlices{vals: []ast.ValueNode{$1}}
+		if $1 == nil {
+			$$ = nil
+		} else {
+			$$ = &valueSlices{vals: []ast.ValueNode{$1}}
+		}
 	}
 	| listElements ',' listElement {
-		$1.vals = append($1.vals, $3)
-		$1.commas = append($1.commas, $2)
-		$$ = $1
+		if $3 == nil {
+			$$ = $1
+		} else if $1 == nil {
+			$$ = &valueSlices{vals: []ast.ValueNode{$3}}
+		} else {
+			$1.vals = append($1.vals, $3)
+			$1.commas = append($1.commas, $2)
+			$$ = $1
```

Comment:
I made these changes when reviewing the nil checks in reverse order: instead of looking at error productions that could produce nil results and then going bottom-up, I was looking at existing nil checks and then looking at their origins, more top-down.

So I added these checks out of an abundance of caution, since there was already a check for a nil `listElements` value [above](https://github.com/bufbuild/protocompile/pull/197/files#diff-cbe13169fa724e83181b277eea5d1e7c0b7b7647441f3330ddb73abf446f0aeeL481). So I added similar check here for consistency. I also went ahead and updated protection against nil `listElement` (singular) values here, too.

However, looking at all of these productions and their dependencies, it's _not_ currently possible to actually get a nil value for either `listElements` or `listElement`. I left these check in anyway. WDYT? Should I remove them if they can't be used?

It might be worth a little re-think on the error handling strategy in here, since it feels a bit brittle. If we do maintenance in this file and add another error production, we could easily reintroduce another nil-dereference panic bug if we weren't cognizant of this potential issue when writing and reviewing the change. ☹️

Reply:
might it depend on the desired way to represent partial parses in general? For example the compound ident node: "a.b.", is that ["a", "b", ""] or ["a", "b"] or ["a", "b", nil]? It also looks like this code is throwing out the comma in the nil case, which I think should be preserved. But then is there an issue in associating which commas separate which values. I think in tree sitter they solve this by having a special ErrorNode, as a place holder.

Reply:
(I'd suggest remove this code for now)

Reply:
> (I'd suggest reverting this code for now)

Ok. Done.